File: CS/SpellCheckerRegistrationHelper.cs
using System;
using System.Text;
using System.Collections.Generic;
using DevExpress.XtraSpellChecker.Native;
using DevExpress.XtraRichEdit.SpellChecker;
using DevExpress.XtraEditors;

namespace SpellCheckerInheritedControls {
    public static class SpellCheckerRegistrationHelper {
        public static void RegisterCustomControlsForSpellChecking() {
            // CustomTextBox
            SpellCheckTextControllersManager.Default.RegisterClass(typeof(CustomTextBox), typeof(SimpleTextBoxTextController));
            SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(typeof(CustomTextBox), typeof(TextBoxFinder));

            // CustomTextEdit
            SpellCheckTextControllersManager.Default.RegisterClass(typeof(CustomTextEdit), typeof(SimpleTextEditTextController));
            SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(typeof(CustomTextEdit), typeof(TextEditTextBoxFinder));

            // CustomMemoEdit
            SpellCheckTextControllersManager.Default.RegisterClass(typeof(CustomMemoEdit), typeof(SimpleTextEditTextController));
            SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(typeof(CustomMemoEdit), typeof(MemoEditTextBoxFinder));

            // CustomRichTextBox
            SpellCheckTextControllersManager.Default.RegisterClass(typeof(CustomRichTextBox), typeof(RichTextBoxTextController));
            SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(typeof(CustomRichTextBox), typeof(RTFTextBoxFinder));

            // CustomRichEditControl
            SpellCheckTextControllersManager.Default.RegisterClass(typeof(CustomRichEditControl), typeof(RichEditSpellCheckController));
            
            // ButtonEdit
            SpellCheckTextControllersManager.Default.RegisterClass(typeof(ButtonEdit), typeof(SimpleTextEditTextController));
            SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(typeof(ButtonEdit), typeof(TextEditTextBoxFinder));

            // ComboBoxEdit
            SpellCheckTextControllersManager.Default.RegisterClass(typeof(ComboBoxEdit), typeof(SimpleTextEditTextController));
            SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(typeof(ComboBoxEdit), typeof(TextEditTextBoxFinder));
        }
    }
}

File: VB/SpellCheckerRegistrationHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Text
Imports System.Collections.Generic
Imports DevExpress.XtraSpellChecker.Native
Imports DevExpress.XtraRichEdit.SpellChecker
Imports DevExpress.XtraEditors

Namespace SpellCheckerInheritedControls
	Public NotInheritable Class SpellCheckerRegistrationHelper
		Private Sub New()
		End Sub
		Public Shared Sub RegisterCustomControlsForSpellChecking()
			' CustomTextBox
			SpellCheckTextControllersManager.Default.RegisterClass(GetType(CustomTextBox), GetType(SimpleTextBoxTextController))
			SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(GetType(CustomTextBox), GetType(TextBoxFinder))

			' CustomTextEdit
			SpellCheckTextControllersManager.Default.RegisterClass(GetType(CustomTextEdit), GetType(SimpleTextEditTextController))
			SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(GetType(CustomTextEdit), GetType(TextEditTextBoxFinder))

			' CustomMemoEdit
			SpellCheckTextControllersManager.Default.RegisterClass(GetType(CustomMemoEdit), GetType(SimpleTextEditTextController))
			SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(GetType(CustomMemoEdit), GetType(MemoEditTextBoxFinder))

			' CustomRichTextBox
			SpellCheckTextControllersManager.Default.RegisterClass(GetType(CustomRichTextBox), GetType(RichTextBoxTextController))
			SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(GetType(CustomRichTextBox), GetType(RTFTextBoxFinder))

			' CustomRichEditControl
			SpellCheckTextControllersManager.Default.RegisterClass(GetType(CustomRichEditControl), GetType(RichEditSpellCheckController))

			' ButtonEdit
			SpellCheckTextControllersManager.Default.RegisterClass(GetType(ButtonEdit), GetType(SimpleTextEditTextController))
			SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(GetType(ButtonEdit), GetType(TextEditTextBoxFinder))

			' ComboBoxEdit
			SpellCheckTextControllersManager.Default.RegisterClass(GetType(ComboBoxEdit), GetType(SimpleTextEditTextController))
			SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(GetType(ComboBoxEdit), GetType(TextEditTextBoxFinder))
		End Sub
	End Class
End Namespace
